xline/src/storage/kvstore.rs

```diff
@@ -35,8 +38,74 @@ pub(crate) struct KvStore {
 /// KV store inner
 #[derive(Debug)]
 struct KvStoreInner {
-    /// B-tree map
-    btree: Mutex<BTreeMap<Vec<u8>, KeyValue>>,
+    /// Key Index
+    index: Mutex<BTreeMap<Vec<u8>, Vec<KeyRevision>>>,
+    /// DB to store key value
+    db: Mutex<HashMap<Revision, KeyValue>>,
+    /// Revision
+    revision: Mutex<i64>,
```

Comment:
Is atomic faster for `i64` type?

Reply:
So far there is only one thread. use mutex is cheap when no contention.